project/ValidatePullRequest.scala

```diff
@@ -3,19 +3,171 @@
  */
 package akka
 
-import sbt._
-import sbt.Keys._
 import com.typesafe.tools.mima.plugin.MimaKeys.reportBinaryIssues
+import net.virtualvoid.sbt.graph.IvyGraphMLDependencies
+import net.virtualvoid.sbt.graph.IvyGraphMLDependencies.ModuleId
+import org.kohsuke.github._
+import sbt.Keys._
+import sbt._
+
+import scala.collection.immutable
+import scala.util.matching.Regex
 
 object ValidatePullRequest extends AutoPlugin {
 
+  val ValidatePR = config("pr-validation") extend Test
+
+  override lazy val projectConfigurations = Seq(ValidatePR)
+
+  /*
+    Assumptions:
+      Env variables set "by Jenkins" are assumed to come from this plugin:
+      https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+request+builder+plugin
+   */
+
+  // settings
+  val PullIdEnvVarName = "ghprbPullId" // Set by "GitHub pull request builder plugin"
+
+  val TargetBranchEnvVarName = "PR_TARGET_BRANCH"
+  val TargetBranchJenkinsEnvVarName = "ghprbTargetBranch"
+  val targetBranch = settingKey[String]("Branch with which the PR changes should be diffed against")
+
+  val SourceBranchEnvVarName = "PR_SOURCE_BRANCH"
+  val SourcePullIdJenkinsEnvVarName = "ghprbPullId" // used to obtain branch name in form of "pullreq/17397"
+  val sourceBranch = settingKey[String]("Branch containing the changes of this PR")
+
+  // asking github comments if this PR should be PLS BUILD ALL
+  val githubEnforcedBuildAll = taskKey[Boolean]("Checks via GitHub API if comments included the PLS BUILD ALL keyword")
+  val buildAllKeyword = taskKey[Regex]("Magic phrase to be used to trigger building of the entire project instead of analysing dependencies")
+
+  // determining touched dirs and projects
+  val changedDirectories = taskKey[immutable.Set[String]]("List of touched modules in this PR branch")
+  val projectIsAffectedByChanges = taskKey[Boolean]("True if this project is affected by the PR and should be rebuilt")
+
+  // running validation
   val validatePullRequest = taskKey[Unit]("Additional tasks for pull request validation")
 
   override def trigger = allRequirements
 
-  override lazy val projectSettings = Seq(
-    validatePullRequest := (),
-    validatePullRequest <<= validatePullRequest.dependsOn(test in Test),
+  override lazy val projectSettings = inConfig(ValidatePR)(Defaults.testTasks) ++ Seq(
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.PerformanceTest"),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.LongRunningTest"),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.TimingTest"),
```

Comment:
When I tried it, it errored out with `Runpath must be either zero or two args`. It seems that it is related [to this issue](https://github.com/scalatest/scalatest/pull/231#issuecomment-40059704) which is fixed in `2.2.0-RC1`. We can update to `2.2.4` in this PR as well.

I also noticed that tests where excluded only when I changed it to `Tests.Argument("-n", "akka.testkit.PerformanceTest")` which is very strange as `-n` should include tags as [ScalaTest documentation](http://www.scalatest.org/user_guide/using_the_runner) suggests. Maybe a bug in ScalaTest sbt framework?